### PR TITLE
OSDOCS: CNV-37922 - Adding docs on VNC token revocation

### PR DESCRIPTION
**Pull request by jherrman**
Version(s):
4.15  and later

Issue:
https://issues.redhat.com/browse/CNV-37922

Link to docs preview:
https://74420--ocpdocs-pr.netlify.app/openshift-dedicated/latest/virt/virtual_machines/virt-accessing-vm-consoles.html
https://74420--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-accessing-vm-consoles.html
https://74420--ocpdocs-pr.netlify.app/openshift-rosa/latest/virt/virtual_machines/virt-accessing-vm-consoles.html

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by akrejcir** [comment]
Looks good, thanks!

/lgtm

**Comment by geetikakay** [comment]
lgtm! Thanks

**Comment by jherrman** [comment]
Reassigning to myself to finish this up, apologies for the delay

File: modules/virt-temporary-token-VNC.adoc [hunk 1]
```diff
@@ -67,3 +68,15 @@ $ oc login --token ${VNC_TOKEN}
...
+
+.Known issue
+
+* Revoking a specific token is not currently possible.
++
+If you require revoking a token, you must delete the service account that was used for creating it. However, this also revokes all other tokens that were created by using the service account.
++
+[source,terminal]
+----
+$ virtctl delete serviceaccount --namespace "<namespace>" "<vm_name>-vnc-access"
```
Comment (bergerhoffer) [inline modules/virt-temporary-token-VNC.adoc:80]: Not sure if virtctl is different than oc, but for oc we normally don't use quotes around arguments

```suggestion
$ virtctl delete serviceaccount --namespace <namespace> <vm_name>-vnc-access
```

Fine to leave as is though if that's right for virtctl

**Review by bergerhoffer — COMMENTED** [review]
A few suggestions and questions!

File: modules/virt-temporary-token-VNC.adoc [hunk 2]
```diff
@@ -67,3 +68,15 @@ $ oc login --token ${VNC_TOKEN}
 ----
 $ virtctl vnc <vm_name> -n <namespace>
 ----
+
+.Known issue
```
[the remark this answers is not in the document]
Comment (jherrman) [reply modules/virt-temporary-token-VNC.adoc:72]: Fair enough, though it seems unnecessarily restrictive (a separate section for known issues or troubleshooting can be eminently useful, for example). Anyhow, will convert it into an admonition.

File: modules/virt-temporary-token-VNC.adoc (hunk 1 above)
Comment (jherrman) [reply modules/virt-temporary-token-VNC.adoc:80]: As per kubevirt docs, the quotation marks should be there - https://github.com/kubevirt/vm-console-proxy/blob/main/docs/api.md#revoking-a-token (possibly to make sure that the values still work if the replaceables are replaced by more complex strings).

**Review by jherrman — COMMENTED** [review]
Thank you for the swift review @bergerhoffer ! On that note, do I understand correctly that you can go ahead and merge the PR when the source changes are ship-shape?

File: modules/virt-temporary-token-VNC.adoc [hunk 3]
```diff
@@ -27,14 +27,15 @@ $ oc patch hyperconverged kubevirt-hyperconverged -n {CNVNamespace} --type json
 # ...
 ----
 
-. Generate a token by running the following command:
+. Generate a token:
```
Comment (mburke5678) [inline modules/virt-temporary-token-VNC.adoc:30]: We now use _by running_ phrasing. See the [doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command).

```suggestion
. Generate a token by running the following command:
```

File: modules/virt-temporary-token-VNC.adoc [hunk 4]
```diff
@@ -27,14 +27,15 @@ $ oc patch hyperconverged kubevirt-hyperconverged -n {CNVNamespace} --type json
...
 +
 [source,terminal]
 ----
 $ curl --header "Authorization: Bearer ${TOKEN}" \
-     "https://api.<cluster_fqdn>/apis/token.kubevirt.io/v1alpha1/namespaces/<namespace>/virtualmachines/<vm_name>/vnc?duration=<duration>" <1>
+     "https://api.<cluster_fqdn>/apis/token.kubevirt.io/v1alpha1/namespaces/<namespace>/virtualmachines/<vm_name>/vnc?duration=<duration>"
```
Comment (mburke5678) [inline modules/virt-temporary-token-VNC.adoc:35]: This structure apparently works if you "comment out the callout(s) in the codeblock". See the [doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks).

```suggestion
     "https://api.<cluster_fqdn>/apis/token.kubevirt.io/v1alpha1/namespaces/<namespace>/virtualmachines/<vm_name>/vnc?duration=<duration>" // <1>
```

File: modules/virt-temporary-token-VNC.adoc [hunk 5]
```diff
@@ -27,14 +27,15 @@ $ oc patch hyperconverged kubevirt-hyperconverged -n {CNVNamespace} --type json
...
 ----
-<1> Duration can be in hours and minutes, with a minimum duration of 10 minutes. Example: `5h30m`. The token is valid for 10 minutes by default if this parameter is not set.
++
```
Comment (mburke5678) [inline modules/virt-temporary-token-VNC.adoc:37]: If you replace the <1>, replace this line ^^

File: modules/virt-temporary-token-VNC.adoc [hunk 6]
```diff
@@ -54,16 +55,28 @@ You can now use the token to access the VNC console of a VM.
 
 .Verification
 
-.  Log in to the cluster by running the following command:
+.  Log in to the cluster:
```
Comment (mburke5678) [inline modules/virt-temporary-token-VNC.adoc:58]:
```suggestion
.  Log in to the cluster by running the following command:
```

File: modules/virt-temporary-token-VNC.adoc [hunk 7]
```diff
@@ -54,16 +55,28 @@ You can now use the token to access the VNC console of a VM.
...
 +
 [source,terminal]
 ----
 $ oc login --token ${VNC_TOKEN}
 ----
 
-.  Use `virtctl` to test access to the VNC console of the VM by running the following command:
+.  Use `virtctl` to test access to the VNC console of the VM:
```
Comment (mburke5678) [inline modules/virt-temporary-token-VNC.adoc:65]:
```suggestion
.  Use `virtctl` to test access to the VNC console of the VM by running the following command:
```

**Comment by mburke5678** [comment]
@jherrman A few suggestions. Otherwise LGTM

**Comment by jherrman** [comment]
@mburke5678 , thank you for the review. When you have a couple minutes, please take a look at the updated doc changes, and let me know if any further modifications are needed for the merge to go ahead.

**Pull request merged**